### DISCUSSION TITLE
Official F1 Telemetry stream: livetiming.formula1.com

### DISCUSSION THREAD

**Discussion by peterkilroy** [theOehrly/Fast-F1#107] [Q&A] [answered]
Hi there,
Really impressive work. Looking forward to v2.2.
I was wondering where one finds further details of the official telemetry stream: livetiming.formula1.com?
I have drawn a bit of a blank and I would like to understand; e.g. How soon after each race is the telemetry updated? Are there any details on the format of the json and json streams?
Thanks for any guidance,
Peter

**Comment by theOehrly** [comment, marked as answer]
There is absolutely no information available that I know of. FastF1s code, the comments within it and the documentation is likely the best you will find in terms of information about the api.

The data is usually available about 30 minutes after a session. But that's just from experience.

If you have any specific questions, feel free to ask.

**Reply by peterkilroy** [reply]
Thanks very much. That's a little disappointing, but on the plus side your source is very well documented.
Is there an obligation to subscribe in order to access the post-race telemetry?

**Reply by theOehrly** [reply]
Hi, sorry, completely forgot to answer. If you want to use live timing/telemetry data visualization in the official f1 app, then you need a subscription. The actual API itself does not require authentication. Therefore, FastF1 does currently not require a subscription.

**Reply by luca-bertero** [reply]
Hi, sorry for the question, but i was wondering why the fast-f1 api doesn't need a subscription for getting the data from livetiming.formula1.com and how it get it.

**Reply by theOehrly** [reply]
The server doesn't ask for authentication, so it's not even possible for FastF1 to authenticate users...

**Reply by EnzoPlayer0ne** [reply]
@theOehrly Hey, first, congratulations on the whole project. It is now quick and easy to analyse F1 data! On top of this, your comments are super helpful.
 I was curious how you reversed the API for 'https://livetiming.formula1.com' if they are no information about it online? 
Also, did you ever have issues with the rate-limiting? For instance, Ergast has a limit of 200 requests per hour. Is there an equivalent for livetimingF1?
Thanks again for the whole project!

**Reply by theOehrly** [reply]
@EnzoPlayer0ne I've never had issues with rate limiting. I don't know if it exists on that API. Usually, I have the caching enabled and that will reduce the number of requests a lot. But even when I had caching disabled, I've never had any problems. If you use FastF1 "reasonably", I'm pretty sure, you won't have any issues.
For finding out, how the API works, there's a few things you can do and then there's plenty of trial and error and guessing for the parsing of the data especially.
